modules/twinklewarn.js

```diff
@@ -1126,6 +1130,11 @@ Twinkle.warn.callback.change_category = function twinklewarnCallbackChangeCatego
 		} );
 	}
 
+	$('select[name=sub_group]').chosen();
+
+	// Limit the max height of select dropdown to prevent dialog box from becoming scrollable
+	mw.util.addCSS('.chosen-container .chosen-results { max-height: 180px !important }');
```

Comment:
This is super thoughtful, but I'm mixed on the end result.  For one, it's around `165px` for me that the background dialog becomes scrollbable/the chosen menu extends beyond the bottom of the warn popup.  Is there a way to replicate what the current menu does, namely extend above (and thus beyond) the dialog?  Some quick checking with `z-index` went nowhere.

Reply:
oh, i tested on 4 browsers, and the results were only slightly different (3-4 px), and settled with 180px. But it looks like it's gonna depend on more factors than just browser.

Reply:
And hey wait a minute, won't this css line affect _all_ chosen dropdowns, including the ones in afch and enterprisey's scripts? That's worrying, ideally chosen should have provided a way to set the max height as an option just like seach_contains.

Reply:
>ideally chosen should have provided a way to set the max height

Yeah, kind of annoying, but hey, free software.

>it would affect other scripts only when the other script is used on the page after Warn was used and before the page is reloaded. Nothing to worry about.

Well, what if someone goes to warn a user but changes their mind?  Unlikely but unfriendly.

Reply:
Thinking about it a bit, couldn't we just grab the drop?  That is:
```suggestion
	$('.chosen-drop').css({'overflow': 'auto', 'max-height': '180px'});
```
When written this way, `180px` doesn't overflow my screen either (not sure why, CSS is weird).

Reply:
This result in 2 scrollables within the chosen drop itself! https://ibb.co/C1bjcTS is what I'm seeing.

Reply:
@siddharthvp  Wha?!😖  I certainly don't get that!  Looks like we're both in chrome atm, have you tried in safemode/a different skin?  Maybe a mac vs pc?  Can you figure out what those two elements are?  Seems weird for both to have the `.chosen-drop` class??

Reply:
The duplicate bars issue notwithstanding, *my* scrollbar is weird with the above; it will disappear below the bottom of the dropdown.  I guessed on `.chosen-drop` but using `.chosen-results` fixes my issue.
```suggestion
	$('.chosen-results').css({'overflow': 'auto', 'max-height': '180px'});
```

Reply:
If we can nail this down I'd like to put it in delsort as well, there's some funkiness going on there with the dropdown extending far below the dialog.

Reply:
`$('.chosen-results').css({'overflow': 'auto', 'max-height': '180px'});` works fine for me too, just like the original mw.util.addCSS fix.

> Can you figure out what those two elements are? Seems weird for both to have the .chosen-drop class??

fwiw the inner one is `div.chosen-results` and the outer is `ul.chosen-drop`

Reply:
So weird, but glad that worked!

Reply:
@siddharthvp Was playing around with adding this to xfd/delsort, but it's a little less clear what is good, though, since the delsort option is lower in the dialog, and a preview would extend the dialog.  135px or 155px seem okay but neither is quite ideal.  Curious what your thoughts are, if it'd even be helpful.

Reply:
@Amorymeltzer I'm getting `.chosen-results` to hit the bottom of the dialog at 147px (with preview off) and at about 167px with preview on. I think we should ignore the latter case, that is, set the max height with the assumption that preview is off. Are you sure its 135px for you?

Reply:
141px actually, but yeah; that's in chrome, I get 142px in Safari and 143px in Firefox.  And for preview, I meant not just clicking the button but not scrolling, but to have the dropdown not overlap the text of the preview.  That for me is about 155px.  Agree with on the 167px case, though fwiw that's my number too.

Reply:
145px is ok for me. But since you said its 141px for you, wouldn't that make the dialog get a scroll bar for you, which is what we're trying to avoid? IMO its okay for the dropdown to overlap the preview text. 

Or maybe its fine to not put in a css rule at all if we're unable to make out what's ideal. The user can scroll down in the dialog to have the full scroll-length of the dropdown visible. It's not much of a issue here as it was in Warn where the select was the most important field.

Reply:
Well, 145px wouldn't be ideal on my end but it certainly wouldn't be a step backward either.

I'm torn — the current view is annoying but context matters in delsort/multi-select and only displaying five could be limiting — so I guess doing nothing is a fine enough place to end up.